tests/unit/test_ansible_inventory.py

```diff
@@ -227,3 +227,189 @@ def test_arbitrary_attrs(self):
+            "no_ca": "yes",
+            "something_else": "for_fun",
+        }
+
+        config = provisioning_config()
+        db = get_db_from_metadata(metadata)
+        ans_inv = AnsibleInventoryOutput(config, db, metadata)
+        inventory = ans_inv.create_inventory()
+
+        srv1 = inventory["all"]["hosts"]["srv1.example.test"]
+
+        assert "readonly_dc" in srv1
+        assert srv1["readonly_dc"] == "yes"
+        assert "something_else" in srv1
+        assert srv1["something_else"] == "not_funny"
+        assert "no_ca" in srv1
+        assert srv1["no_ca"] == "no"
+
+        srv2 = inventory["all"]["hosts"]["srv2.example.test"]
+        assert "no_ca" in srv2
+        assert "something_else" in srv2
+        assert srv2["no_ca"] == "yes"
+        assert srv2["something_else"] == "for_fun"
+
+    def test_global_arbitrary_attrs(self):
+        """
+        Test that values defined in `ansible_inventory` dictionary in the
+        job metadata file gets into host attributes in generated ansible
+        inventory.
+        """
+        metadata = metadata_extra()
+        metadata["ansible_inventory"] = {
+            "no_ca": "no",
+            "something_else": "not_funny",
+        }
+
+        config = provisioning_config()
+        db = get_db_from_metadata(metadata)
+        ans_inv = AnsibleInventoryOutput(config, db, metadata)
+        inventory = ans_inv.create_inventory()
+
+        srv1 = inventory["all"]["hosts"]["srv1.example.test"]
+
+        assert "readonly_dc" not in srv1
+        assert "something_else" in srv1
+        assert srv1["something_else"] == "not_funny"
+        assert "no_ca" in srv1
+        assert srv1["no_ca"] == "no"
+
+        srv2 = inventory["all"]["hosts"]["srv2.example.test"]
+        assert "no_ca" in srv2
+        assert "something_else" in srv2
+        assert srv2["no_ca"] == "no"
+        assert srv2["something_else"] == "not_funny"
+
+    def test_global_arbitrary_attrs_domain_override(self):
+        """
+        Test that values defined in `ansible_inventory` dictionary in the
+        job metadata file gets into host attributes in generated ansible
+        inventory.
+        """
+        metadata = metadata_extra()
+        metadata["ansible_inventory"] = {
+            "no_ca": "no",
+            "something_else": "not_funny",
+        }
+
+        metadata["domains"][0]["ansible_inventory"] = {
+            "no_ca": "yes",
+        }
+
+        config = provisioning_config()
+        db = get_db_from_metadata(metadata)
+        ans_inv = AnsibleInventoryOutput(config, db, metadata)
+        inventory = ans_inv.create_inventory()
+
+        srv1 = inventory["all"]["hosts"]["srv1.example.test"]
+
+        assert "readonly_dc" not in srv1
+        assert "something_else" in srv1
+        assert srv1["something_else"] == "not_funny"
+        assert "no_ca" in srv1
+        assert srv1["no_ca"] == "yes"
+
+        srv2 = inventory["all"]["hosts"]["srv2.example.test"]
```

Comment:
I think it would be nice to run this test with 2 domains, one host in each of them. Test that global scope attributes are set on the host (when they are not defined at domain level), and test that domain scope attributes have higher precedence than global scope ones.

I.e. Define an attribute at global level, and have 2 domains with one of them overwriting the global scope one.

Reply:
I will add this particular test case

Reply:
Test added